### PR TITLE
feat(api): adjust link return DTOs

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#378

### Dependencies

- Upstream: none
- Downstream:
  - https://github.com/metriport/metriport-internal/pull/454
  - https://github.com/metriport/metriport/pull/162

### Description

[Context/thread about the issues driving this PR](https://metriport.slack.com/archives/C04DMKE9DME/p1679099450027649).

- adjust link return DTOs to match API and SDK
- update SDK JSDoc
- export JSDoc on SDK package
- standardize SDK error message when missing API data
- remove CW person ID if we get a 404 when getting link
- include the whole person name when returning CW links
- gender at birth:
  - API input: `M | F`
  - API output: `M | F | U` (for links, `U` being unspecified)

### Release Plan

- nothing special, just merge along dependencies

### CODE REVIEW

File: packages/packages/api/src/medical/models/patient.ts [hunk 1]
```diff
@@ -52,6 +12,7 @@ export type PatientUpdate = z.infer<typeof patientUpdateSchema>;
 
 export const patientSchema = patientUpdateSchema.extend({
   facilityIds: z.array(z.string()),
+  links: linkStatusAcrossHIEsSchema,
```
Comment (Goncharo) [inline packages/packages/api/src/medical/models/patient.ts:15]: Why do we need `links` in the patient update? Updating a patient's ID/demo is currently separate from any notion of links

File: packages/packages/api/src/medical/models/link.ts [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
-  patient: Patient;
-}
+export const patientLinksSchema = z.object({
+  potentialLinks: linkSchema.array(),
+  currentLinks: linkSchema.array(),
+});
+export type PatientLinks = z.infer<typeof patientLinksSchema>;
+
+export const linkStatusSchema = z.literal("linked").or(z.literal("needs-review"));
+export type LinkStatus = z.infer<typeof linkStatusSchema>;
```
Comment (Goncharo) [inline packages/packages/api/src/medical/models/link.ts:23]: Need to update the docs accordingly

**Review by Goncharo — CHANGES_REQUESTED** [review]

File: packages/packages/api/README.md [hunk 3]
```diff
@@ -4,23 +4,46 @@ A package to get started using [Metriport](https://metriport.com/) - universal a
...
+
+```ts
+new MetriportMedicalApi("YOUR_API_KEY", { sandbox: true });
 ```
-import { Metriport } from "@metriport/api";
 
-const metriportClient = new Metriport("YOUR_API_KEY", "ENVIRONMENT_URL");
+#### Devices API
+
+To use the Medical API, create a new instance of the `MetriportDevicesApi` class:
```
Comment (Goncharo) [inline packages/packages/api/README.md:35]: nit: To use the Devices API

File: packages/packages/api/README.md [hunk 4]
```diff
@@ -4,23 +4,46 @@ A package to get started using [Metriport](https://metriport.com/) - universal a
...
+```ts
+import { MetriportMedicalApi } from "@metriport/api";
+
+const metriportClient = new MetriportMedicalApi("YOUR_API_KEY");
+```
+
+To connect to the sandbox:
+
+```ts
+new MetriportMedicalApi("YOUR_API_KEY", { sandbox: true });
```
Comment (Goncharo) [inline packages/packages/api/README.md:30]: Awesome, was thinking about doing this for a while now haha - need to update the docs accordingly though, right now we still have them passing the URL

**Review by Goncharo — CHANGES_REQUESTED** [review]

File: packages/packages/api/src/medical/models/link.ts (hunk 2 above)
Comment (leite08) [reply packages/packages/api/src/medical/models/link.ts:23]: done

File: packages/packages/api/README.md (hunk 3 above)
Comment (leite08) [reply packages/packages/api/README.md]: done

File: packages/packages/api/README.md (hunk 4 above)
Comment (leite08) [reply packages/packages/api/README.md:30]: done

**Comment by leite08** [comment]
@Goncharo, I've sent a [new commit](https://github.com/metriport/metriport/pull/161/commits/49e35d96d7d6a4f1bb6e2edeba2eb563e9340ec2) with a bunch of changes, mostly on the docs.

File: docs/medical-api/api-reference/patient/list-patients.mdx [hunk 5]
```diff
@@ -12,8 +12,9 @@ api: "GET /medical/v1/patient"
 
 ## Response
 
-<ResponseField type="Patient[]">
-  All Patients receiving care at the specified Facility.
+All Patients receiving care at the specified Facility.
+
+<ResponseField name="patients" type="Patient[]">
```
Comment (Goncharo) [inline docs/medical-api/api-reference/patient/list-patients.mdx:17]: Need to update the `type="Patient[]"` here, and all the places where we list factilities/patients/docs

It's not returning `Patient[]`, but `{patients: Patient[]}`

My bad for not doing this properly to begin with, I was referencing the SDK returns when documenting, not the endpoints

**Review by Goncharo — CHANGES_REQUESTED** [review]

File: docs/medical-api/api-reference/patient/list-patients.mdx (hunk 5 above)
Comment (leite08) [reply docs/medical-api/api-reference/patient/list-patients.mdx:17]: But this is saying there's a property with a field named `patients`, which is an array.

Do you want to have a wrapper around this to say its a JSON?

Comment (Goncharo) [reply docs/medical-api/api-reference/patient/list-patients.mdx:17]: Just noticed the `name="patients"` -> can ignore this!

**Review by Goncharo — APPROVED** [review]
🚢 🚢 🚢 🚢

**Pull request merged**